lang/java/reef-runtime-local/src/main/java/org/apache/reef/runtime/local/process/RunnableProcessState.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.local.process;
+
+/**
+ * The possible states of a process: INIT, RUNNING, ENDED.
+ */
+enum RunnableProcessState {
+
+  /** After initialization. */
+  INIT,
+
+  /** The process is running. */
+  RUNNING,
+
+  /** The process ended. */
+  ENDED;
+
+  /**
+   * Check whether a transition from current state to the given one is legal.
+   * @param toState destination state.
+   * @return True if the state transition is legal, false otherwise.
+   */
+  public boolean isLegal(final RunnableProcessState toState) {
+
+    switch (this) {
```

Comment:
Why is transition from `INIT` to `INIT` legal, but transitions from `RUNNING` to `RUNNING` and from `ENDED` to `ENDED` illegal? Seems suspicious to me